src/iso15118/io/socket_helper.cpp

```diff
@@ -32,20 +33,30 @@ bool get_first_sockaddr_in6_for_interface(const std::string& interface_name, soc
             continue;
         }
 
-        if (interface_name.compare(current_if->ifa_name) != 0) {
+        if (interface_name.compare("auto") != 0 && interface_name.compare(current_if->ifa_name) != 0) {
             continue;
         }
 
         // NOTE (aw): because we did the check for AF_INET6, we can assume that ifa_addr is indeed an sockaddr_in6
-        memcpy(&address, current_if->ifa_addr, sizeof(address));
+        const sockaddr_in6* current_addr = reinterpret_cast<const sockaddr_in6*>(current_if->ifa_addr);
```

Comment:
would use `const auto socket_addr = ` here, as the type is already explicitly stated in the `reinterpret_cast`.

Reply:
You are right. I will change that